schemaexpr: plumb proper eval.Context into deserializeExprForFormatting

yuzefovich opened the pull request:
**schemaexpr: plumb proper eval.Context into deserializeExprForFormatting**

Long time ago in 781b3aca893f18fe7aca6cb1e25eea89dacef0cc we introduced a call to `expr.Eval` with using empty `eval.Context` in `deserializeExprForFormatting`. However, we found at least one case where this results in an internal error (when evaluating a CHECK constraint that uses datetime builtin functions like `parse_time` which need non-empty `TxnTimestamp` field). This commit addresses that issue by plumbing the proper eval context (the only exception is `sql.ValidateConstraint` method where plumbing seemed non-trivial).

Fixes: #97643.

**sql: remove PlanHookState.RunParams**

This method is unusual that it returns unexported `runParams` struct, so this commit makes a minor refactor to remove the method from the interface.

Release note: None

yuzefovich committed:
schemaexpr: plumb proper eval.Context into deserializeExprForFormatting

Long time ago in 781b3aca893f18fe7aca6cb1e25eea89dacef0cc we introduced
a call to `expr.Eval` with using empty `eval.Context` in
`deserializeExprForFormatting`. However, we found at least one case where
this results in an internal error (when evaluating a CHECK constraint that
uses datetime builtin functions like `parse_time` which need non-empty
`TxnTimestamp` field). This commit addresses that issue by plumbing the
proper eval context (the only exception is `sql.ValidateConstraint`
method where plumbing seemed non-trivial).

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/124618)
<!-- Reviewable:end -->

yuzefovich committed:
sql: remove PlanHookState.RunParams

This method is unusual that it returns unexported `runParams` struct, so
this commit makes a minor refactor to remove the method from the
interface.

Release note: None

yuzefovich commented:
Friendly ping @rafiss

rafiss reviewed (approved):
thanks for the fix! lgtm

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/124618#-:-NzUirmdDiIHYeh1kU6w:btfbvdc)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained



<!-- Sent from Reviewable.io -->

yuzefovich commented:
TFTR!

bors r+
